packages/backend/src/core/transaction-count/RpcTransactionUpdater.ts

```diff
@@ -90,6 +90,8 @@ export class RpcTransactionUpdater implements TransactionCounter {
   async update() {
     this.logger.info('Update started', { project: this.projectId.toString() })
 
+    await this.blockQueue.waitTilEmpty()
```

Comment:
this deserves a long comment on why it's so important to wait until the queue is empty :)